java/src/main/java/org/rocksdb/DBOptions.java

```diff
@@ -403,22 +403,6 @@ public int maxBackgroundJobs() {
     return maxBackgroundJobs(nativeHandle_);
   }
 
-  @Override
-  @Deprecated
-  public DBOptions setMaxBackgroundCompactions(
```

Comment:
options like `max_background_compactions` is deprecated but not removed yet in c++ API. I think we should keep them in sync.

Reply:
@jay-zhuang will those C++ deprecated methods be removes for RocksDB 7? Is there any sort of list of which will and which won't be removed that you can share with us please?

Reply:
Here is list of options/api we removed, I think the java API is already updated:
* max_mem_compaction_level
* soft_rate_limit
* hard_rate_limit
* rate_limit_delay_max_milliseconds
* purge_redundant_kvs_while_flush
* base_background_compactions
* skip_log_error_on_recovery
* preserve_deletes
* iter_start_seqnum
* table_cache_remove_scan_count_limit
* new_table_reader_for_compaction_inputs
* Remove deprecated overloads of DB::AddFile()
* Remove DB::CompactRange() marked as DEPRECATED_FUNC

Some other deprecated APIs may not be able to remove in this release.